### PR TITLE
adds CE and steps timing INFO logs

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

**Comment by simonbrandhof** [comment]
LGTM

File: sonar-core/src/main/java/org/sonar/core/util/logs/DefaultProfiler.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import org.sonar.api.utils.System2;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.LoggerLevel;
+
+class DefaultProfiler extends Profiler {
+
+  private static final String CONTEXT_SEPARATOR = " | ";
+
+  private final LinkedHashMap<String, Object> context = new LinkedHashMap<>();
+  private final Logger logger;
```
Comment (SonarTech) [inline sonar-core/src/main/java/org/sonar/core/util/logs/DefaultProfiler.java]: MAJOR Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1312)

File: server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java [hunk 2]
```diff
@@ -125,22 +136,36 @@ private void populateFromModules() {
    * and lazily starts its components.
    */
   private static MutablePicoContainer createContainer(ComponentContainer parent) {
-    ReflectionLifecycleStrategy lifecycleStrategy = new ReflectionLifecycleStrategy(new NullComponentMonitor(), "start", "stop", "close") {
+    ComponentMonitor componentMonitor = instanceComponentMonitor();
+    ReflectionLifecycleStrategy lifecycleStrategy = new ReflectionLifecycleStrategy(componentMonitor, "start", "stop", "close") {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java]: MAJOR Make this a named "static" inner class. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

File: server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java [hunk 3]
```diff
@@ -125,22 +136,36 @@ private void populateFromModules() {
...
+  private static ComponentMonitor instanceComponentMonitor() {
+    if (!LOG.isTraceEnabled()) {
+      return new NullComponentMonitor();
+    }
+    return new ComputeEngineComponentMonitor();
+  }
+
+  private static class ComputeEngineComponentMonitor extends NullComponentMonitor {
+
+    public <T> void instantiated(PicoContainer container, ComponentAdapter<T> componentAdapter,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/container/ComputeEngineContainerImpl.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 3 major

Watch the comments in this conversation to review them.

**Pull request merged**